### ISSUE TITLE
Mail version 4

### ISSUE THREAD

**Issue by rompic** [herrbischoff/awesome-osx-command-line#136]
I had to set the mail_version to "V4" on macOS 10.12.3

**Comment by herrbischoff** [comment]
Good catch! I will amend the entry accordingly.

**Comment by qu1gl3s** [comment]
I got it working with V4 for macOS 10.12.3 with this change.

```applescript
tell application "Mail" to quit
set os_version to do shell script "sw_vers -productVersion"
set mail_version to "V2"
considering numeric strings
	if "10.10" ≤ os_version then set mail_version to "V3"
	if "10.12" ≤ os_version then set mail_version to "V4"
end considering

set sizeBefore to do shell script "ls -lnah ~/Library/Mail/" & mail_version & "/MailData | grep -E 'Envelope Index$' | awk {'print $5'}"
do shell script "/usr/bin/sqlite3 ~/Library/Mail/" & mail_version & "/MailData/Envelope\\ Index vacuum"

set sizeAfter to do shell script "ls -lnah ~/Library/Mail/" & mail_version & "/MailData | grep -E 'Envelope Index$' | awk {'print $5'}"

display dialog ("Mail index before: " & sizeBefore & return & "Mail index after: " & sizeAfter & return & return & "Enjoy the new speed!")

tell application "Mail" to activate```

I've also lodged a merge request with this change having been made.

**Comment by herrbischoff** [comment]
Merged #140 just now which is the first pull request correctly implementing this.
